259 pass component

AnSavvides opened the pull request:
Closes #259 - this passes `component` to `prefixer` so that we can use its name for warnings. I have updated all relevant tests & examples. I also bumped up the range for `max-params` in `. eslintrc`.

Any feedback is very welcome!

AnSavvides commented:
Updated `karma` & `karma-webpack` to get the tests running on AppVeyor & Travis as expected.

ianobermiller commented on modules/keyframes.js:
```diff
@@ -27,6 +27,7 @@ if (isAnimationSupported) {
 // Simple animation helper that injects CSS into a style object containing the
 // keyframes, and returns a string with the generated animation name.
 var keyframes = function (
+  component: any,
```
I don't think keyframes should have to take this in; let`s just pass the component as null, since you may not have a component to pass (e.g. you export a bunch of keyframe animations from a module).

AnSavvides replied:
Good shout, I read your first sentence and then immediately thought about what you are suggesting! :smile:

ianobermiller commented:
Nice! More work than I expected, thanks for taking this on!

AnSavvides commented:
No worries, I am more than happy to help :)

I moved the `component` param so that it can be optional, updated the tests, added another test that confirms you can use keyframes without a component and finally updated the docs + examples.

ianobermiller commented on docs/api/README.md:
```diff
@@ -117,10 +117,12 @@ Usage:
 
 Create a keyframes animation for use in any inline style. `keyframes` is a helper that translates the keyframes object you pass in to CSS and injects the `@keyframes` (prefixed properly) definition into a style sheet. Automatically generates and returns a name for the keyframes, that you can then use in the value for `animation`. Radium will automatically apply vendor prefixing to keyframe styles ...
 
+`Radium.keyframes` takes a second optional parameter, `component`. This is optional as you may not always have a component to pass. If you do have a `component` however, it is a good idea to pass that as a parameter for better warning & error reporting.
```
nice summary!

ianobermiller commented on modules/__mocks__/prefixer.js:
```diff
@@ -8,7 +8,7 @@ module.exports = {
     return name;
   },
-  getPrefixedStyle: style => style,
+  getPrefixedStyle: (component, style) => (component, style),
```
the second `(component, style)` doesn't do anything, should just be `style`

AnSavvides replied:
:+1:

ianobermiller commented on modules/__tests__/keyframes-test.js:
```diff
@@ -45,7 +49,7 @@ describe('keyframes', () => {
     expect(document.head.appendChild).not.to.have.been.called;
 
-    var name = keyframes({});
+    var name = keyframes({}, MyComponent);
```
can you make a copy of a simple test where you don't pass the component, just to make sure it doesn't explode?

AnSavvides replied:
I have already done that [here](https://github.com/AnSavvides/radium/commit/e31e89a58124b4711328afadac1b017045a457b2#diff-af1a46446733ef5ab0fcce554afc942fR82).

ianobermiller replied:
shouldnt the test have failed, then? :S

AnSavvides replied:
It shouldn't, since `component` is optional, right? :confused:

ianobermiller commented on modules/prefixer.js:
```diff
@@ -385,13 +386,16 @@ var getPrefixedStyle = function (style, mode /* 'css' or 'js' */) {
-        console.warn('Unsupported CSS property "' + property + '"');
+        console.warn(
+          'Unsupported CSS property "' + property +
+          '" in component "' + component.constructor.displayName + '"'
```
this will fail if `component` is null

AnSavvides replied:
Ah yes, that wouldn't be fun :) I will add a check to see whether we have a component or not.

AnSavvides commented:
Thanks for the feedback @ianobermiller :)

ianobermiller commented on modules/prefixer.js:
```diff
@@ -385,13 +389,19 @@ var getPrefixedStyle = function (style, mode /* 'css' or 'js' */) {
+          : '';
+
+        console.warn(
+          'Unsupported CSS property "' + property + componentContext
```
Missing quotes if no component. Remove it from above and do this here (using template strings for clarity):

`Unsupported CSS property "${property}"` + componentContext

AnSavvides replied:
:+1:

ianobermiller commented:
Sorry, just one more small change :)

AnSavvides commented:
No worries, it's definitely better to get it right than push out breaking changes or bad code :)
